src/librustc_mir/interpret/step.rs

```diff
@@ -132,7 +132,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     ///
     /// There is no separate `eval_rvalue` function. Instead, the code for handling each rvalue
     /// type writes its results directly into the memory specified by the place.
-    fn eval_rvalue_into_place(
+    pub fn eval_rvalue_into_place(
```

Comment:
Oh, why this? We're trying to keep a clean interface for the interpret, that'll never work if everyone just adds `pub` whenever they feel like it. :(

Reply:
Looking at the const_prop changes, it seems like what happened is that const_prop moved up a layer or two of the interpreter abstractions. Shouldn't that mean that while some new operations need to be public now, some of the previously used one can be made private? If you could scan the functions that const_prop no longer needs and make them private, that would make me much less concerned about this change. :)

Reply:
> Shouldn't that mean that while some new operations need to be public now, some of the previously used one can be made private? 

Yes, I think you're correct. I'll do that in the other PR as well.